### PR TITLE
feat: failover on bind error

### PR DESCRIPTION
**Pull request by Jakub-Vacek**
## Description

This PR adds ability to switch providers when bind is unsuccessful (related to https://www.notion.so/superface/Mails-are-not-send-from-Brain-7462ed7c349c45e8940749bbd0cd5c48#9bbf5ebde54e4dbaa5c54caa36fb6279). If the overall approach is ok I will add/fix tests and handle all resolution cases

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have updated the documentation accordingly. If you are changing **code related to user secrets** you need to really make sure that [security documentation](SECURITY.md) is correct.
- [x] I have read the **CONTRIBUTION_GUIDE** document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by TheEdward162 — COMMENTED** [review]
I think this is fine

**Comment by Jakub-Vacek** [comment]
I switched to bypassing the policy logic - in case of circuit breaker it would not switch. 
In case of error during bind we should switch to second provider if we can - providerFailover is true and we have second provider.

**Comment by TheEdward162** [comment]
@Jakub-Vacek I think it would be better if the router could react to the error kind instead. So we'd add a new error variant to https://github.com/superfaceai/one-sdk-js/blob/d737f542aeb05fcda733e2d7dbdd95a4ee8b33f8/src/client/failure/policy.ts#L57 and then the policy would react to it by temporarily setting the number of retries to zero (e.g. switching right away). What do you think?

**Comment by Jakub-Vacek** [comment]
Well, my solution is hacky, yours feels less hacky :D I will try it over the weekend.

**Comment by Jakub-Vacek** [comment]
@TheEdward162 weekend is over (😅) and I tried to rewrite it according to your comment.

File: src/client/failure/policies.ts [hunk 1]
```diff
@@ -264,6 +264,9 @@ export class RetryPolicy extends FailurePolicy {
   }
 
   override afterFailure(info: ExecutionFailure): FailureResolution {
+    if (info.kind === 'bind') {
+      this.streak = 0;
```
Comment (TheEdward162) [inline src/client/failure/policies.ts:268]: Shouldn't this set the streak to maximum to force an abort?

Comment (Jakub-Vacek) [reply src/client/failure/policies.ts]: Yep, i was handling this also in CircuitBreaker so this code never got used. Now it is handled in inner policy.

**Review by TheEdward162 — APPROVED** [review]

**Pull request merged**